src/Configuration/Annotation/Embedded.php

```diff
@@ -4,16 +4,23 @@
 
 namespace Hateoas\Configuration\Annotation;
 
+use JMS\Serializer\Annotation\AnnotationUtilsTrait;
+use Symfony\Contracts\Service\Attribute\Required;
+
 /**
  * @Annotation
  * @Target("ANNOTATION")
  */
+#[\Attribute(0)]
```

Comment:
What's the purpose of passing "0"?

(the same question applies to all usages of `#[\Attribute(0)]` in this PR

Reply:
I tried to resolve an issue that may not be an issue.  I intended not to allow the attribute to be used on any of the existing targets. It seems the `0` produces this behaviour.
```
Attribute::TARGET_CLASS
Attribute::TARGET_FUNCTION
Attribute::TARGET_METHOD
Attribute::TARGET_PROPERTY
Attribute::TARGET_CLASS_CONSTANT
Attribute::TARGET_PARAMETER
Attribute::TARGET_ALL
```

For example, `Embedded` can be used only as a nested attribute.

I can remove the `0` for `Embedded`, `Exclusion` and `Route` as I am not sure if it could be a problem for certain configurations.